session/session.go

```diff
@@ -1198,6 +1198,21 @@ func (s *session) Auth(user *auth.UserIdentity, authentication []byte, salt []by
 	user.AuthUsername, user.AuthHostname, success = pm.ConnectionVerification(user.Username, user.Hostname, authentication, salt)
 	if success {
 		s.sessionVars.User = user
+		// Load default active roles if login successed.
```

Comment:
Could this be moved to somewhere like `session.go:loadCommonGlobalVariablesIfNeeded()` ?

Could the result be cached? Image that 1K connection connect to tidb simultaneously, tikv hot region will be a bottleneck for this operation.

Reply:
Ok, I will cache it.